blueoil/cmd/output_event.py

```diff
@@ -63,32 +74,42 @@ def output(tensorboard_dir, output_dir, metrics_keys, steps, output_file_base="m
 
     columns = [_column_name(event_accumulator, metrics_key)
                for event_accumulator, metrics_key in itertools.product(event_accumulators, metrics_keys)]
-    columns.sort()
-    df = pd.DataFrame([], columns=columns)
+
+    values_step_dict = {}
+    value_matrix = []
+
+    for metrics_key in metrics_keys:
+        if not value_matrix:
+            step_list = sorted(_step_list(event_accumulator, metrics_key), reverse=True)
+            value_matrix.append(step_list)
 
     for event_accumulator in event_accumulators:
         for metrics_key in metrics_keys:
             value_step_list = _value_step_list(event_accumulator, metrics_key)
-            for value, step in value_step_list:
-                column_name = _column_name(event_accumulator, metrics_key)
-                df.loc[step, column_name] = value
+            values_step_dict = dict(value_step_list)
 
-    if steps:
-        df = df[steps, :]
-
-    df = df.sort_index(ascending=False)
-
-    # index to column. and re-order column.
-    df["step"] = df.index
-    df = df[["step"] + columns]
+            for step in step_list:
+                if step not in values_step_dict:
+                    values_step_dict[step] = ''
+            sorted_value_step = OrderedDict(sorted(values_step_dict.items(), key=lambda x: x[0], reverse=True))
+            sorted_list = list(sorted_value_step.values())
+            value_matrix.append(sorted_list)
+    data_by_row = list(map(list, zip(*value_matrix)))
+    columns.insert(0, "step")
 
     output_csv = os.path.join(output_dir, "{}.csv".format(output_file_base))
-    df.to_csv(output_csv, index=False)
+
+    with open(output_csv, "w") as fp:
+        wr = csv.writer(fp)
+        wr.writerow(columns)
+        for row in data_by_row:
```

Comment:
Can we use `writerows` ?
https://docs.python.org/ja/3/library/csv.html#csv.csvwriter.writerows

Reply:
Thank you! It's a good suggestion, I have changed it.